tools/codegen/codespec/merge_attributes.go

```diff
@@ -0,0 +1,79 @@
+package codespec
+
+import "sort"
+
+func mergeAttributes(pathParams, createRequest, createResponse, readResponse Attributes) Attributes {
+	merged := make(map[string]*Attribute)
+
+	addOrUpdate := func(attr *Attribute, computability ComputedOptionalRequired) {
+		if existingAttr, exists := merged[attr.Name]; exists {
+			if existingAttr.Description == nil || *existingAttr.Description == "" {
+				existingAttr.Description = attr.Description
+			}
+
+			if existingAttr.ComputedOptionalRequired == Required {
+				return
+			}
+
+			existingAttr.ComputedOptionalRequired = computability
+		} else {
+			newAttr := *attr
+			newAttr.ComputedOptionalRequired = computability
+			merged[attr.Name] = &newAttr
+		}
+	}
+
+	// Path parameters: all attributes will be "required"
+	for i := range pathParams {
```

Comment:
This might not work. E.g., `projectId` is a path parameter but it shouldn't be required

Reply:
not sure i understand, if it's a path parameter it will be required no?
Can you share a use-case?

Reply:
https://github.com/mongodb/atlas-sdk-go/blob/main/openapi/atlas-api-transformed.yaml#L1150

Reply:
hmmm, that's true. For example for cluster, groupId and clusterId can be path params in read/GET but in the schema only groupId will be a required attribute and clusterId would actually be computed.

I think then a general approach here can be only create/POST path params should be required.

 @AgustinBettati does that sound good? or I may be missing a case? lmk will update the spec then

Reply:
I agree with the approach of defining as required only the path params present in the POST operation. If there are other path params in GET/UPDATE/DELETE the will very likely be present in the response of the post (computed).